mail-manager/src/main/scala/it/pagopa/interop/commons/mail/MailData.scala

```diff
@@ -0,0 +1,46 @@
+package it.pagopa.interop.commons.mail
+
+import javax.mail.internet.InternetAddress
+import org.typelevel.literally.Literally
+import scala.util.Try
+
+sealed trait Mail {
```

Comment:
Just wondering if a simple discriminator field could be an alternative solution.
Mail implementation al exactly the same.

Reply:
It really depends on how do you plan to expand this feature in the future. If we want to add other kinds of mails, probably oop is better since we may add fields that do no adhere to the trait. I'll think a bit about it.